src/Symfony/Component/String/AbstractString.php

```diff
@@ -646,6 +646,11 @@ public function truncate(int $length, string $ellipsis = ''): self
      */
     abstract public function upper(): self;
 
+    /**
+     * Returns the printable length on a terminal.
+     *
+     * If the string contains a non-printable character, -1 is returned.
```

Comment:
That's the only thing I'm unsure about. While this matches `wcswidth()`, is this the most useful behavior?
Should be drop any control chars before calling `wcswidth()`?
`$s = preg_replace('/\p{Cc}++/u', '', $s);`

Reply:
I think we can change it. width() is already not 100% wcswidth compliant.

Reply:
yes, let's do this
since the main use case is to compute column width in the console, we could check how those chars are (not) displayed there.